asnake/aspace/__init__.py

```diff
@@ -0,0 +1,138 @@
+			# Not sure if this is safe
+			if attr.lower().endswith("s"):
+				shortCalls = ["repositories", "locations", "subjects", "users", "vocabularies", "location_profiles", "container_profiles"]
+				#for calls without repositories in them
+				if attr in shortCalls:
+					return jsonmodel_muliple_object(self.__client.get("/" + str(attr) + "?all_ids=true").json(), self.__client, self.repository, attr)
+				else:
+					return jsonmodel_muliple_object(self.__client.get("/repositories/" + str(self.repository) + "/" + str(attr) + "?all_ids=true").json(), self.__client, self.repository, attr)
+
+	# not sure if theres a way to pass a variable to implement this with __getattr__
+	def resource(self, id):
+		return jsonmodel_single_object(self.__client.get("repositories/" + self.repository + "/resources/" + str(id)).json(), self.__client)
+	
+	#this doesn't work yet
+	def resourceID(self, id):
+		result = self.__client.get("/repositories/" + self.repository + "/search?page=1&aq={\"query\":{\"field\":\"identifier\", \"value\":\"" + str(id) + "\", \"jsonmodel_type\":\"field_query\"}}").json()
+		resourceURI = result["results"][0]["uri"]
+		return jsonmodel_single_object(self.__client.get(resourceURI).json(), self.__client)		
+		
+	def archival_object(self, id):
+		if isinstance(id, str):
+			if len(id) == 32:
+				# its a ref_id
+				params = {"ref_id[]": str(id)}
+				refList = self.__client.get("repositories/" + self.repository + "/find_by_id/archival_objects?page=1&ref_id[]=" + str(id)).json()
+				return jsonmodel_single_object(self.__client.get(refList["archival_objects"][0]["ref"]).json(), self.__client)
+		#its a uri number
+		return jsonmodel_single_object(self.__client.get("repositories/" + self.repository + "/archival_objects/" + str(id)).json(), self.__client)
+	
+	def agents(self, type, id = None):
+		if id == None:
+			return jsonmodel_muliple_object(self.__client.get("/agents/" + str(type) + "?all_ids=true").json(), self.__client, self.repository, type)
+		else:
+			return jsonmodel_single_object(self.__client.get("/agents/" + str(type) + "/" + str(id)).json(), self.__client)
+
+		
+class jsonmodel_single_object:
+	def __init__(self, json_rep, client = None):
+		self.__json = json_rep
+		self.__client = client
+	
+	def __getattr__(self, key):
+	
+		if isinstance(key, str):
+			if not key.startswith('_'):
+				if not key in self.__json.keys():
+					repository = self.__json["repository"]["ref"].split("/repositories/")[1]
+					return jsonmodel_muliple_object(self.__json, self.__client)
+		
+		if isinstance(self.__json[key], list):
+			return jsonmodel_muliple_object(self.__json[key], self.__client)
+		elif isinstance(self.__json[key], dict):
+			return jsonmodel_single_object(self.__json[key], self.__client)
+		else:
+			return self.__json[key]
+
+	# utility methods
+	def pp(self):
+		print (json.dumps(self.__json, indent=2))
+
+	def json(self):
+		return self.__json
+
+	def serialize(self, filePath):
```

Comment:
Hmmm. I possibly cut serialize or replace with something that serializes to a string?  I feel like a conversion to string should be in the class, and the caller should handle writing that string to a file/file-like thing?

Reply:
Yeah, that makes sense, if we even need that, If we do add stuff like that it probably should be in a more optional and less important import.